subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/MavenComponentParser.java

```diff
@@ -233,130 +235,94 @@ private static boolean dependencyMatchesProject(ModuleDependency dependency, Mod
         return coordinates.getModule().equals(DefaultModuleIdentifier.newId(dependency.getGroup(), dependency.getName()));
     }
 
-    private static List<SoftwareComponentVariant> getSortedVariants(SoftwareComponentInternal component) {
+    private static Stream<? extends SoftwareComponentVariant> getSortedVariants(SoftwareComponentInternal component) {
```

Comment:
Why do you think this is odd? `UsageContext` is an outdated type, and we wouldn't be using it at all if KMP didn't depend on it. `UsageContext` and `SoftwareComponentVariant` are pretty much identical. In general we should be avoiding `UsageContext` if possible

Reply:
I expect sort to return a complete collection, not a `Stream`.  It's a little weird conceptually to have a sorted `Stream`, since the whole point of a stream or sequence is to operate on things one at a time and sorted/not sorted is a binary property of the entire collection.  

I also wouldn't expect sorting to change the element type at all - including getting less specific.

And in general, returning an unterminated `Stream` from a method not named `.stream()` instead of calling a terminal operation and returning a collection is unusual.  Makes sense to avoid the extra work in some cases, like here, but rarely the norm.

Reply:
This method is used in two places, so I would prefer not to duplicate the logic. 
The goal here was to avoid allocating a list when it is just going to get iterated over. Granted, that will likely be negligible for performance, but I see no reason why this isn't a suitable approach

Is there a better name for the method you think would be more appropriate?

Reply:
I really don't think it's a big deal to inline in 2 methods, especially if it's stream-building code with the important sorting logic already extracted to its own method.  But if you want to leave it like this, maybe name it `createSortedVariantsStream`?  I think that would be a little better.

Reply:
I don't feel there is anything wrong with returning a `Stream`. I did rename this to `createSortedVariantsStream` however